worker/task/process_retry_event_delivery.go

```diff
@@ -238,14 +238,40 @@ func ProcessRetryEventDelivery(endpointRepo datastore.EndpointRepository, eventD
 			requestLogger.Errorf("%s", eventDelivery.UID)
 			done = false
 
-			eventDelivery.Status = datastore.RetryEventStatus
-
-			nextTime := time.Now().Add(delayDuration)
-			eventDelivery.Metadata.NextSendTime = nextTime
-			attempts := eventDelivery.Metadata.NumTrials + 1
+			// Get subscription to check delivery mode
+			subscription, err := subRepo.FindSubscriptionByID(ctx, eventDelivery.ProjectID, eventDelivery.SubscriptionID)
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Error("failed to find subscription")
+				return &EndpointError{Err: err}
+			}
 
-			log.FromContext(ctx).Errorf("%s next retry time is %s (strategy = %s, delay = %d, attempts = %d/%d)\n", eventDelivery.UID,
-				nextTime.Format(time.ANSIC), eventDelivery.Metadata.Strategy, eventDelivery.Metadata.IntervalSeconds, attempts, eventDelivery.Metadata.RetryLimit)
+			// For at-most-once delivery, only retry on network failures
+			if subscription.DeliveryMode == datastore.AtMostOnceDeliveryMode {
+				if resp.StatusCode < 100 {
```

Comment:
Consider extracting this logic so it's easier to test in isolation and change if needed. 
```suggestion
				if retriableForAtMostOnceDeliveryMode(resp.StatusCode) {
```